src/include/daos_fs_sys.h

```diff
@@ -0,0 +1,492 @@
+/**
+ * (C) Copyright 2018-2021 Intel Corporation.
+ *
+ * SPDX-License-Identifier: BSD-2-Clause-Patent
+ */
+/**
+ * \file
+ *
+ * DAOS File System "Sys" API
+ *
+ * The DFS Sys API provides a simplified layer directly on top
+ * of the DFS API that is more similar to the equivalent
+ * POSIX libraries. While the DFS Sys API stands on its own,
+ * the DFS API can be used directly by getting the DFS Object
+ * with dfs_sys2base().
+ */
+
+#include <dirent.h>
+
+#include "daos.h"
+#include "daos_fs.h"
+
+#define DFS_SYS_NO_CACHE 1
+#define DFS_SYS_NO_LOCK 2
+
+/** struct holding attributes for the dfs_sys calls */
+typedef struct dfs_sys dfs_sys_t;
+
+/**
+ * Mount a file system with dfs_mount and optionally
+ * initialize a cache.
+ *
+ * \param[in]	poh	Pool connection handle.
+ * \param[in]	coh	Container open handle.
+ * \param[in]	mflags	Mount flags (O_RDONLY or O_RDWR).
+ * \param[in]	sflags	Sys flags (DFS_SYS_NO_CACHE or DFS_SYS_NO_LOCK)
+ * \param[out]	dfs_sys	Pointer to the file system object created.
+ *
+ * \return		0 on success, errno code on failure.
+ */
+int
+dfs_sys_mount(daos_handle_t poh, daos_handle_t coh, int mflags, int sflags,
+	      dfs_sys_t **dfs_sys);
```

Comment:
Shall we go beyond that and also hide the pool/container connect/open?

Reply:
Do you mean pass the pool and container UUID instead of open handles, and dfs_sys_mount would open the pool/container?
If so, I have a few concerns/points:
- if the user wants to mount several containers in the same pool, then the pool would be re-opened each time
- it's possible that the user wants finer control over the handles. For example, in mpifileutils we first open a container to see if it's POSIX before calling dfs_mount. And dfs_mount itself also takes open handles. This is necessary because the datamover tools switch between the DFS, Object, and HDF5 APIs
But we could do something like:
```
dfs_sys_mount(...)
{
    daos_cont_open(...)
    daos_cont_query(...)
    if not POSIX
        return ERR
    dfs_mount(...)
}
```

Thoughts @mchaarawi ?

Reply:
I will make a proposal for DFS API for an option to hide these, and we can extend that to DFS_sys after.